extensions/Khronos/KHR_mesh_compression/README.md

```diff
@@ -0,0 +1,172 @@
+* Frank Galligan, Google, <mailto:fgalligan@google.com>
+* Kai Ninomiya, Google, <mailto:kainino@google.com>
+* Patrick Cozzi, Cesium, [@pjcozzi](https://twitter.com/pjcozzi)
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension defines an schema to use Draco geometry compression libraries in glTF format. This allows glTF to support streaming compressed geometry data instead of the raw data.
+
+The [conformance](#conformance) section specifies what an implementation must to do when encountering this extension, and how the extension interacts with the attributes defined in the base specification.
+
+## glTF Schema Updates
+
+Draco geometry compression library could be used for `primitive` by adding an `extension` property to a primitive, and defining its `KHR_draco_geometry_compression` property.
+
+The following picture shows the structure of the schema update. 
+
+**Figure 1**: Structure of geometry compression extension.
+![](figures/structure.png)
+
+In general, we will use the extension to point to the buffer that contains the compressed data. The major change is that `accessor` in extended `primitive` no
+longer point to a `bufferView`. Instead, the `attributes` of a primitive will use the decompressed data. This is valid because in glTF 2.0, `bufferView` is not required in `accessor`, although if it is not present or the id is 0, it will be used with `sparse` field to act as a sparse accessor. In this extension, we will ignore the `bufferView` property.
+
+Usage of the extension must be listed in the `extensionUsed` and `extensionsRequired`. 
+
+```javascript
+"extensionsUsed" : [
+    "KHR_draco_geometry_compression"
+]
+
+"extensionsRequired" : [
+    "KHR_draco_geometry_compression"
+]
+
+```
+
+The extension then could be used like:
+
+```javascript
+
+"mesh" : {
+    "primitives" : [
+        {
+            "attributes" : {
+                "POSITION" : 11,
+                "NORMAL" : 12,
+                "TEXCOORD_0" : 13,
+            },
+            "indices" : 10,
+            "mode" : 4
+            "extensions" : {
+                "KHR_draco_geometry_compression" : {
+                    "buffer" : 10,
+                    "byteOffset" : 1024,
+                    "byteLength" : 10000,
+                    "indicesCount" : 1000,
+                    "vertexCount" : 500,
+                    "geometryType" : "mesh",
+                    "attributesOrder" : [
+                        "POSITION",
+                        "NORMAL",
+                        "TEXCOORD_0"
+                    ],
+                    "attributesComponentTypes" : [
+                        "VEC3",
+                        "VEC3",
+                        "VEC2"
+                    ],
+                    "attributesType" : [
+                        5126,
+                        5126,
+                        5126
+                    ]
+                }
```

Comment:
`attributesComponentTypes` and `attributesType` must be switched to match core spec definitions.

`attributesOrder`, `attributesComponentTypes`, and `attributesType` arrays must be of the same size and must list elements in the same order, right? So why not combine these properties into one structure:

```json
{
    "attributes" : [
        {
            "semantic": "POSITION",
            "type": "VEC3",
            "componentType": 5126
        },
        {
            "semantic": "NORMAL",
            "type": "VEC3",
            "componentType": 5126
        },
        {
            "semantic": "TEXCOORD_0",
            "type": "VEC2",
            "componentType": 5126
        },
    ]
}
```

`type: "VEC3"` and `componentType: 5126` could be made default, since we could expect them to be the most frequent values.